src/graphmap.rs

```diff
@@ -166,21 +174,40 @@ where
     }
 }
 
-impl<N, E, Ty> GraphMap<N, E, Ty>
+impl<N, E, Ty, S> GraphMap<N, E, Ty, S>
 where
     N: NodeTrait,
     Ty: EdgeType,
+    S: BuildHasher,
 {
     /// Create a new `GraphMap`
-    pub fn new() -> Self {
+    pub fn new() -> Self
+    where
+        S: Default,
+    {
```

Comment:
FYI: by leaving `S` open here, it means that type inference can't figure it out, and it won't use the default either. e.g. This used to work:

```rust
let mut graph = GraphMap::new();
// ... other code that insert stuff, inferring those types but *not* the hasher
```

Now to allow inference I have to write:

```rust
let mut graph = GraphMap::<_, _, _>::new(); // now 4th param `S` is defaulted
```

That's why `HashMap::new()` and `IndexMap::new()` only use the default hasher, while `default()` is fully generic.
See also: https://faultlore.com/blah/defaults-affect-inference/

Reply:
Don't get me wrong, this is a great feature overall! I would have just left `new()` on the old hasher for the reason above, but it would be a breaking change to go back on that now.

Reply:
Ack, thanks for the comment. I would have done it differently if I'd realized.